### PR TITLE
Bump `FFMpegRecipe` and `PyAVRecipe` versions

### PR DESCRIPTION
**Pull request by DexerBR**
`PyAV` updated to the latest stable release - `13.1.0`
`FFmpeg` updated to `6.1.2` - `ffpyplayer` is still not compatible with FFmpeg `7.x.x.`

### CODE REVIEW

File: pythonforandroid/recipes/av/__init__.py [hunk 1]
```diff
@@ -5,11 +5,15 @@
...
-    version = "10.0.0"
+    version = "13.0.0"
     url = "https://github.com/PyAV-Org/PyAV/archive/v{version}.zip"
 
     depends = ["python3", "cython", "ffmpeg", "av_codecs"]
     opt_depends = ["openssl"]
+    patches = ['patches/compilation_syntax_errors.patch']
+
+    def prebuild_arch(self, arch):
+        self.apply_patches(arch)
```
Comment (AndreMiras) [inline pythonforandroid/recipes/av/__init__.py]: I think this is pretty much what's happening already, so we can probably remove that override and let it apply the patches automatically.
https://github.com/kivy/python-for-android/blob/v2024.01.21/pythonforandroid/build.py#L495-L497

**Review by AndreMiras — DISMISSED** [review]
Thanks for taking care of it.
Left a minor comment and let's see how the build is doing in the CI

File: pythonforandroid/recipes/av/__init__.py (hunk 1 above)
Comment (DexerBR) [reply pythonforandroid/recipes/av/__init__.py]: That makes sense, removed!

File: pythonforandroid/recipes/ffmpeg/patches/configure.patch
Comment (DexerBR) [inline pythonforandroid/recipes/ffmpeg/patches/configure.patch:1]: I had a false positive, but this patch is still needed, and a patch for libx264 is now also needed.

File: pythonforandroid/recipes/ffmpeg/__init__.py [hunk 2]
```diff
@@ -52,7 +51,7 @@ def build_arch(self, arch):
                 build_dir = Recipe.get_recipe(
                     'libx264', self.ctx).get_build_dir(arch.arch)
                 cflags += ['-I' + build_dir + '/include/']
-                ldflags += ['-lx264', '-L' + build_dir + '/lib/']
+                ldflags += [build_dir + '/lib/' + 'libx264.a']
```
Comment (DexerBR) [inline pythonforandroid/recipes/ffmpeg/__init__.py]: Recent versions of ffmpeg seem to prioritize searching for a dynamic library, ignoring the static one unless explicitly defined. The solution, then, is to provide the path to the static library directly.

Comment (AndreMiras) [reply pythonforandroid/recipes/ffmpeg/__init__.py]: Great findings, do you mind putting this exact comment above that line in the source?

Comment (DexerBR) [reply pythonforandroid/recipes/ffmpeg/__init__.py]: Good ideia, done! 👍

**Comment by AndreMiras** [comment]
Hi @DexerBR, thank you for looking this up.
I noticed the build log has a lot of theses while building av.
```
av/_core.pyx:5:3: cimported module has no attribute 'avdevice_register_all'
...
av/_core.pyx:54:25: cimported module has no attribute 'swresample_configuration'
```
See for instance the [Test updated recipes for arch x86_64 [ ubuntu-latest ]](https://github.com/kivy/python-for-android/actions/runs/11063369869/job/30739400548?pr=3065) build.
Do you have an idea of why this could be?

**Comment by DexerBR** [comment]
> Hi @DexerBR, thank you for looking this up. I noticed the build log has a lot of theses while building av.
> 
> ```
> av/_core.pyx:5:3: cimported module has no attribute 'avdevice_register_all'
> ...
> av/_core.pyx:54:25: cimported module has no attribute 'swresample_configuration'
> ```
> 
> See for instance the [Test updated recipes for arch x86_64 [ ubuntu-latest ]](https://github.com/kivy/python-for-android/actions/runs/11063369869/job/30739400548?pr=3065) build. Do you have an idea of why this could be?

They usually used to be errors related to premature compilation issues. The patch I'm using, for example, solves the same type of errors, but caused by problems recognizing more modern Python syntax (I don't know why).

In this specific case, the error seems to have originated somewhere else, some tips could be:

```
[INFO]:    -> running python3 setup.py build_ext -v --ffmpeg-dir=/home/u...(and 90 more)
Exception in thread background thread for pid 57736:
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/user/app/venv/lib/python3.10/site-packages/sh.py", line 1639, in wrap
    fn(*rgs, **kwargs)
  File "/home/user/app/venv/lib/python3.10/site-packages/sh.py", line 2641, in background_thread
    handle_exit_code(exit_code)
  File "/home/user/app/venv/lib/python3.10/site-packages/sh.py", line 2332, in fn
    return self.command.handle_command_exit_code(exit_code)
  File "/home/user/app/venv/lib/python3.10/site-packages/sh.py", line 826, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1: 

  RAN: /home/user/.local/share/python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build/python3 setup.py build_ext -v --ffmpeg-dir=/home/user/.local/share/python-for-android/build/other_builds/ffmpeg/x86_64__ndk_target_24/ffmpeg

  STDOUT:
/home/user/.local/share/python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build/Lib/site-packages/_distutils_hack/__init__.py:11: UserWarning: Distutils was imported before Setuptools, but importing Setuptools also replaces the `distutils` module in `sys.modules`. This may lead to undesirable behaviors or errors. To avoid these issues, avoid using distutils directly, ensure that setuptools is installed in the traditional way (e.g. not an editable install), and/or make sure that setuptools is always imported before distutils.
  warnings.warn(
/home/user/.local/share/python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build/Lib/site-packages/_distutils_hack/__init__.py:26: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
Compiling av/_core.pyx because it changed.
[1/1] Cythonizing av/_core.pyx
Compiling av/stream.pyx because it changed.
```

and 

```
[1/1] Cythonizing av/audio/frame.pyx
Traceback (most recent call last):
  File "/home/user/.local/share/python-for-android/build/other_builds/av-openssl/x86_64__ndk_target_24/av/setup.py", line 216, in <module>
    setup(
  File "/home/user/.local/share/python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build/Lib/site-packages/setuptools/__init__.py", line 104, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/.local/share/python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build/Lib/site-packages/setuptools/_distutils/core.py", line 172, in setup
    ok = dist.parse_command_line()
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/.local/share/python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build/Lib/site-packages/setuptools/_distutils/dist.py", line 475, in parse_command_line
    args = self._parse_command_opts(parser, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/.local/share/python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build/Lib/site-packages/setuptools/dist.py", line 869, in _parse_command_opts
    nargs = _Distribution._parse_command_opts(self, parser, args)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/.local/share/python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build/Lib/site-packages/setuptools/_distutils/dist.py", line 541, in _parse_command_opts
    raise DistutilsClassError(
distutils.errors.DistutilsClassError: command class <class 'setuptools.command.build_ext.build_ext'> must subclass Command


  STDERR:


[INFO]:    av first build failed (as expected)
[INFO]:    Running cython where appropriate
[INFO]:    Cythonize av/_core.pyx
stty: 'standard input': Inappropriate ioctl for device
[INFO]:    -> running python3 -cimport sys; from Cython.Compiler.Main im...(and 65 more)
Exception in thread background thread for pid 57744:
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/user/app/venv/lib/python3.10/site-packages/sh.py", line 1639, in wrap
    fn(*rgs, **kwargs)
  File "/home/user/app/venv/lib/python3.10/site-packages/sh.py", line 2641, in background_thread
    handle_exit_code(exit_code)
  File "/home/user/app/venv/lib/python3.10/site-packages/sh.py", line 2332, in fn
    return self.command.handle_command_exit_code(exit_code)
  File "/home/user/app/venv/lib/python3.10/site-packages/sh.py", line 826, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1: 
```

Before the patch, the errors I mentioned occurred in these same places.

<br>

So in short, these errors break further compilation resulting in the errors you mentioned.

**Review by AndreMiras — APPROVED** [review]
Hey @DexerBR, thank you for taking the time to clarify and I could also confirm it building the recipe in its current version from develop and got the same error (see https://github.com/AndreMiras/python-for-android/actions/runs/11386543386/job/31679029568).
We're good to :ship: it
